metron-interface/metron-alerts/src/app/alerts/meta-alerts/meta-alerts.component.ts

```diff
@@ -0,0 +1,101 @@
+import {UpdateService} from '../../service/update.service';
+import {SearchRequest} from '../../model/search-request';
+import {SearchService} from '../../service/search.service';
+import {SearchResponse} from '../../model/search-response';
+import {SortField} from '../../model/sort-field';
+import {META_ALERTS_INDEX} from '../../utils/constants';
+import {AlertSource} from '../../model/alert-source';
+import {PatchRequest} from '../../model/patch-request';
+import {Patch} from '../../model/patch';
+
+@Component({
+  selector: 'app-meta-alerts',
+  templateUrl: './meta-alerts.component.html',
+  styleUrls: ['./meta-alerts.component.scss']
+})
+export class MetaAlertsComponent implements OnInit {
+
+  selectedMetaAlert = '';
+  searchResponse: SearchResponse = new SearchResponse();
+
+  constructor(private router: Router,
+              private metaAlertService: MetaAlertService,
+              private updateService: UpdateService,
+              private searchService: SearchService) {
+  }
+
+  goBack() {
+    this.router.navigateByUrl('/alerts-list');
+    return false;
+  }
+
+  ngOnInit() {
+    let searchRequest = new SearchRequest();
+    searchRequest.query = '*';
+    searchRequest.from = 0;
+    searchRequest.size = 999;
+    searchRequest.facetFields = [];
+    searchRequest.indices =  [META_ALERTS_INDEX];
+    searchRequest.sort = [new SortField('threat:triage:score', 'desc')];
+    searchRequest.fields = ['id', 'alert_status', 'threat:triage:score', 'count', 'guid', 'name'];
+
+    this.searchService.search(searchRequest).subscribe(resp => this.searchResponse = resp);
+  }
+
+  doAddAlertToMetaAlert(alertSources: AlertSource[]) {
+    let patchRequest = new PatchRequest();
+    patchRequest.guid = this.selectedMetaAlert;
+    patchRequest.sensorType = 'metaalert';
+    patchRequest.index = META_ALERTS_INDEX;
+    patchRequest.patch = [new Patch('replace', 'alert', alertSources)];
+
+    this.updateService.patch(patchRequest).subscribe(rep => {
+      console.log('Meta alert saved');
+      this.goBack();
+    });
+  }
+
+  addAlertToMetaAlert() {
+    let searchRequest = new SearchRequest();
+    searchRequest.query = 'guid:"' + this.selectedMetaAlert + '"';
+    searchRequest.from = 0;
+    searchRequest.size = 1;
+    searchRequest.facetFields = [];
+    searchRequest.indices =  [META_ALERTS_INDEX];
+    searchRequest.sort = [];
+    searchRequest.fields = [];
+
+    this.searchService.search(searchRequest).subscribe((searchResponse: SearchResponse) => {
+      if (searchResponse.results.length === 1) {
+        searchResponse.results[0].source.alert = [...searchResponse.results[0].source.alert,
+                                                  ...this.metaAlertService.selectedAlerts.map(alert => alert.source)];
+        this.doAddAlertToMetaAlert(searchResponse.results[0].source.alert);
+      } else {
+        console.log('Unable to get a single meta alert');
+      }
+    });
+    console.log(this.selectedMetaAlert);
```

Comment:
should this be here?

Reply:
As mentioned in the Next Section of CC, we need some kind of notification once actions like add/remove are done otherwise it is confusing. These are placeholders till we have a notification mechanism. If it is not a problem I would suggest we leave them else I can remove them.